cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers;
+
+import co.cask.cdap.api.security.store.SecureStore;
+import co.cask.cdap.api.security.store.SecureStoreData;
+import co.cask.cdap.api.security.store.SecureStoreManager;
+import co.cask.cdap.common.BadRequestException;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import co.cask.cdap.proto.security.SecureStoreCreateRequest;
+import co.cask.http.HttpResponder;
+import com.google.inject.Inject;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+
+import java.io.IOException;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * Exposes REST APIs for {@link co.cask.cdap.api.security.store.SecureStore} and
+ * {@link co.cask.cdap.api.security.store.SecureStoreManager}.
+ */
+@Path(Constants.Gateway.API_VERSION_3 + "/security/store/namespaces/{namespace-id}")
```

Comment:
Partitioning by namespace was not part of the design but we had decided to keep it as part of the exposed API so we can do the partitioning trivially if required.